### ISSUE TITLE
Host dialog: rework host and session settings

### ISSUE THREAD

**Issue by askmeaboutlo0m** [drawpile/Drawpile#1220]
The host and session settings dialogs have grown pretty unwieldy and need to be resdesigned. Here's some things that need to be adressed, it needs to be fleshed out further and a UI thought up for it:

* The host dialog must make the distinction between public and private sessions more explicit, rather than having it be implicit via setting a password. This causes confusion all the time. It should default to private.

* The host and session settings dialog should just be a single dialog, so that you don't first have to host a session and then go into the settings to finish configuring it. If a server doesn't allow a setting, like persistence, you can just get a message in chat about it.

* Operator password and session password need to be separated more clearly, probably into different pages. People often get them mixed up.

* There's currently session templates, permission templates, role exports and ban exports. You have to import all of these separately if you want to host a session, which takes a ridiculous number of clicks. There should be a single export for them and when you import them you should be able to pick which pieces you want to import.

**Comment by askmeaboutlo0m** [comment]
To kick off the discussion on this maybe, here's an attempt at scribbling out the design of the new dialog.

I'm not particularly attached to this design, it's just something that came to mind. Feedback (both in favor of and against things) and alternate ideas are desired.

The idea is that this dialog would become the Host page in the start dialog. The separate session settings dialog would go away. The menu option for it would continue to exist, but it just sends you to the Host page instead.

The main idea behind this change is that it's just less clicking around to set up a session properly and less cognitive load of having to learn your way around two different dialogs.

The separate import and export for permissions, roles and bans would go away, instead, there'll be a unified import and export. Clicking on the export button would offer you a set of checkboxes asking you what you want to export. The import functionality should still be able to handle the old formats.

I'll go through the individual tabs of it to explain the design ideas and what issues they're supposed to address.

### Session

There's now a toggle front and center that makes the distinction between public and private sessions clear. When you check Private, the Password field is shown and typing in a password is mandatory. When you attempt to toggle it while the session is running, it will ask you to confirm that you want to do that.

The Password field is only shown if the session is set to Public. The Generate button would be a Change button when you're connected to the session I guess.

"Terminate session when everyone has left" is equivalent to the old "persist without users", just the other way round. I think this is a more sensible way to word it to avoid someone thinking that making a session persistent will make it persist *forever* and then get sad because it ran into the idle timeout and their drawings are lost. If a server doesn't allow persistent sessions, the user will get a warning after hosting.

The selection of where to host is now a single drop-down that lets you pick the server. When you do, it will attempt to load information about that server from some kind of predefined address, giving server owners the chance to provide a description and a link to their rules right here. It would also allow communicating permissions, like if the server allows persistent sessions, what the idle timeout is like etc. It should integrate with the list server in a sensible way so that we can use the existing Servers settings and not have two separate systems going on. The integration should be optional, someone without a list server should be able to just dump a JSON file with an appropriate format somewhere. The drop-down would have the following options:

* The servers you've hosted on or joined before.
* Servers provided by list servers you have added (which would be a new list server feature.)
* `pub.drawpile.net` as a hard-coded option.
* "Add or remove servers", letting you add more options here.
* "Browse community servers", which will provide an in-application view of the communities page of drawpile.net, giving them more visibility. This would likely be a separate task.
* "Host server on this computer", which will replace the information text with a techy warning about port forwarding and whatnot, hopefully scaring away users from picking this option if they don't know what they're doing.

The "advanced options" that used to be here have been banished to the Listing tab instead.

![Session tab, hosting](https://github.com/drawpile/Drawpile/assets/13625824/6315364e-279e-471b-9c07-41d317d18744)

When the session is running, the server selection instead becomes this set of settings. While you could set these before hostin the session, we don't really want users to do that, it's up to the server to decide the defaults for these.

![Session tab, running session](https://github.com/drawpile/Drawpile/assets/13625824/4ea9f399-474f-4313-a7f8-5e7e0c8fe709)

### Permissions

The login permission stuff is here now. The "block new logins" option is not shown here because it's slated to be removed, see #1222. It would still show up if someone actually enables it though, to let you disable it again. After which it would disappear again.

The feature permissions are presented in a table format and slightly reworded. Maybe there's a more sensible way to order them, but I couldn't think of any.

![Permissions tab](https://github.com/drawpile/Drawpile/assets/13625824/e53c4411-d60e-403f-9ad3-3b63f12f2c73)

### Roles

Mostly the same as the old page, but the operator password is here now, to prevent it from getting mixed up with the session password. When it's assigned, the Assign button would change to Change instead.

![Roles tab](https://github.com/drawpile/Drawpile/assets/13625824/90b8b92c-b720-4e0a-8313-e09146bb43b8)

### Listing

The ID alias field is here now, since it's kinda part of how the session is listed.

When you host a session, you can put listings in here in advance, rather than how it was before where you could only put a single listing into the dialo and then had to go into the session settings to add more.

The private listing stuff that used to be here is gone because it's subject to removal in #1224.

![Listing tab](https://github.com/drawpile/Drawpile/assets/13625824/840f9a1b-53a6-48d3-bc67-c4ffeb94981f)

### Bans

Basically the same as it was.

![Bans tab](https://github.com/drawpile/Drawpile/assets/13625824/0aade341-7a1c-4bd9-ad06-77b02852b126)

**Comment by askmeaboutlo0m** [comment]
Another note (credit kale): the Event Log has a Users tab that shows very similar information to the Roles tab and it's pretty confusing to a user why those are separate and how they differ. They both show past and present users, so it might make more sense to unify them into a single table that just shows everything.
